Assets/Scripts/Core/Match/GameController.cs

```diff
@@ -31,48 +40,72 @@ private void Construct(SignalBus signalBus, PlayerData.Factory playerFactory, Pl
         private void Awake()
         {
             MainGUID = AddPlayer(_playerSettings.Colors.Dequeue()).GUID;
-            SceneController.SceneLoaded += OnSceneLoaded;        
+            SceneController.SceneLoaded += OnSceneLoaded;
+            foreach(var player in _players.Values)
+            {
+                _addPrana.Add(player.GUID, 0);
+            }
+            _cities = _mapController.Cities;
         }
         private void Start()
         {
             _signalBus.Subscribe<GameStartedSignal>(OnGameStarted);
-            _signalBus.Subscribe<PlayerBuildingTempleSignal>(OnPlayerBuildingTemple);
-            _signalBus.Subscribe<AddPranaSignal>(AddPrana);
-            _signalBus.Subscribe<RemovePranaSignal>(RemovePrana);
-            _signalBus.Subscribe<SacrificeAcceptedSignal>(SacrificeAccepted);
+            _signalBus.Subscribe<PlayerTryBuildTempleSignal>(OnPlayerBuildingTemple);
+            _signalBus.Subscribe<AddPranaSignal>(OnAddPrana);
+            _signalBus.Subscribe<RemovePranaSignal>(OnRemovePrana);
+            _signalBus.Subscribe<SacrificeAcceptedSignal>(OnPlayerAcceptedSacrifice);
+        }
+        private void Update()
+        {
+            AddPassivePrana();
+        }
+        private void AddPassivePrana()
+        {
+            if (_timer > 0f) _timer -= Time.deltaTime;
+            else
+            {
+                foreach (var city in _cities)
+                {
+                    List<AddPranaSignal> collectPrana = city.CollectPrana();
+                    for (int i = 0; i < collectPrana.Count; i++)
+                    {
+                        if (_addPrana.ContainsKey(collectPrana[i].PlayerGUID))
+                            _addPrana[collectPrana[i].PlayerGUID] += collectPrana[i].Value;
+                    }
+                }
+                Dictionary<string, byte> addPranaWithNoZeroValue = _addPrana.Where(x => x.Value > 0).ToDictionary(i => i.Key, i => i.Value);
+                foreach (var prana in addPranaWithNoZeroValue)
+                {
+                    OnAddPrana(new AddPranaSignal { PlayerGUID = prana.Key, Value = prana.Value});
+                }
+                ResetAddPrana();
+                _timer = 1f;
+            }
+        }
```

Comment:
Почему бы просто не пройтись по каждому игроку и не просуммировать прану с каждого подконтрольного города?
```suggestion
        private void AddPassivePrana()
        {
            if (_timer > 0f) _timer -= Time.deltaTime;
            else
            {
                int sum = 0;               
                foreach (string playerGUID in _players.Keys)
                {
                    var cities = _mapController.SelectMany(city => !city.IsNeutral).ByOwner(playerGUID);
                    foreach (var city in cities) sum += city.CollectPrana();                 
                }
                _timer = 1f;
            }
        }
```

Reply:
`var cities = _mapController.SelectMany(city => !city.IsNeutral).ByOwner(playerGUID);`
Я же не просто так писал методы расширения 😄 Рекомендую пользоваться `SelectMany()`, `ByDistance()`, `ByOwner()` и т.п.